### PR TITLE
Add MULTISAMPLED_SHADING downlevel flag

### PR DESCRIPTION
**Pull request by jinleili**
**Connections**
About https://github.com/gfx-rs/wgpu/issues/2419

### CODE REVIEW

File: wgpu-hal/src/vulkan/adapter.rs [hunk 1]
```diff
@@ -929,13 +932,13 @@ impl super::Instance {
...
-            );
-            return None;
-        }
+        // if phd_features.core.sample_rate_shading == 0 {
+        //     log::warn!(
+        //         "sample_rate_shading feature is not supported, hiding adapter: {}",
+        //         info.name
+        //     );
+        //     return None;
+        // }
```
Comment (cwfitzgerald) [inline wgpu-hal/src/vulkan/adapter.rs:941]: If this isn't needed, we should just remove it.

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
One nit, but otherwise LGTM

File: wgpu-hal/src/vulkan/adapter.rs [hunk 2]
```diff
@@ -342,6 +344,7 @@ impl PhysicalDeviceFeatures {
             Df::FRAGMENT_WRITABLE_STORAGE,
             self.core.fragment_stores_and_atomics != 0,
         );
+        dl_flags.set(Df::MULTISAMPLED_SHADING, self.core.sample_rate_shading != 0);
```
Comment (cwfitzgerald) [inline wgpu-hal/src/vulkan/adapter.rs:347]: The other backends should also set this appropriately

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
Sorry missed one thing

File: wgpu-hal/src/vulkan/adapter.rs (hunk 2 above)
Comment (jinleili) [reply wgpu-hal/src/vulkan/adapter.rs:347]: But I didn't find sample_rate_shading logic at other backends

Comment (cwfitzgerald) [reply wgpu-hal/src/vulkan/adapter.rs:347]: If there is no flag determining if it is supporting, they should always enable it.

Comment (jinleili) [reply wgpu-hal/src/vulkan/adapter.rs:347]: Checked other backends, all flags are enabled by default like Vulkan backends：
https://github.com/gfx-rs/wgpu/blob/b19000367c15ebc3f4aa51871e6d6b9b670126f8/wgpu-hal/src/metal/adapter.rs#L1036

Comment (cwfitzgerald) [reply wgpu-hal/src/vulkan/adapter.rs:347]: Ahhh I didn't catch that. Cool

**Review by cwfitzgerald — APPROVED** [review]
Thanks!

**Pull request merged**